src/vocab_encoder.cpp

```diff
@@ -61,14 +63,14 @@ bool VocabEncoder::evaluate(ov::TensorVector& outputs, const ov::TensorVector& i
     outputs[0].set_shape({num_elements});
     auto token_ids = outputs[0].data<int32_t>();
 
-    for (size_t element_idx = 0; element_idx < num_elements; ++element_idx) {
-        auto element = m_vocab->find(std::vector<uint8_t>(chars + begins[element_idx], chars + ends[element_idx]));
+    ov::parallel_for(num_elements, [&](size_t element_idx){
+        const auto element = m_vocab->find(std::string(chars + begins[element_idx], chars + ends[element_idx]));
         if (element == m_vocab->end()) {
             token_ids[element_idx] = default_value;
         } else {
             token_ids[element_idx] = element->second;
         };
```

Comment:
please check if we can remove if-else and use default_value for all non-existent key.